packages/datatrak-web-server/src/routes/RecentSurveysRoute.ts

```diff
@@ -0,0 +1,40 @@
+/**
+ * Tupaia
+ * Copyright (c) 2017 - 2023 Beyond Essential Systems Pty Ltd
+ */
+
+import { Request } from 'express';
+import camelcaseKeys from 'camelcase-keys';
+import { Route } from '@tupaia/server-boilerplate';
+import { DatatrakWebRecentSurveysRequest } from '@tupaia/types';
+
+export type RecentSurveysRequest = Request<
+  DatatrakWebRecentSurveysRequest.Params,
+  DatatrakWebRecentSurveysRequest.ResBody,
+  DatatrakWebRecentSurveysRequest.ReqBody,
+  DatatrakWebRecentSurveysRequest.ReqQuery
+>;
+
+export class RecentSurveysRoute extends Route<RecentSurveysRequest> {
+  public async buildResponse() {
+    const { query, models } = this.req;
+
+    const { userId } = query;
+
+    // The central server queries already built in don't support GROUP BY and this would be unnecessary to add for this one use case, so we use raw SQL in this instance
+    const surveyResponses = await models.database.executeSql(
```

Comment:
I think it's fair enough to go around the central server if it doesn't support group by and I agree that would be out of scope to add support for. Apart from that we should use [parameter binding](https://knexjs.org/guide/raw.html#raw-parameter-binding) to pass in the userId since it is  a value coming over the network and could be a security risk.

Also I have a personal preference for using the model syntax over raw sql but I think that is just personal preference. @EMcQ-BES would you also mind casting your eye over this in case we have forgotten anything. For example, do we need to consider permissions in this context?

Reply:
@tcaiger We raised the permissions question with Georgia. Technically a user should _always_ have permissions for surveys they themselves submitted (or else they couldn't have submitted them)

The edge case of someone having their permissions revoked after submitting we've agreed the tile will still display, but upon clicking through the user will be rejected from the survey route.

Reply:
As far as the model syntax, we can't actually use a `group by` with it either. The `buildQuery` function in `TupaiaDatabase` doesn't have support for it. When we need a group by we always either use `RAW` or run the grouping in js

Reply:
Thanks Ethan. So then the only change we need to make here is the param binding / replacements tweak and then it's good to go